storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java

```diff
@@ -0,0 +1,578 @@
+import static org.apache.kafka.server.log.remote.storage.LocalTieredStorageEvent.EventType.FETCH_OFFSET_INDEX;
+import static org.apache.kafka.server.log.remote.storage.LocalTieredStorageEvent.EventType.FETCH_PRODUCER_SNAPSHOT;
+import static org.apache.kafka.server.log.remote.storage.LocalTieredStorageEvent.EventType.FETCH_SEGMENT;
+import static org.apache.kafka.server.log.remote.storage.LocalTieredStorageEvent.EventType.FETCH_TIME_INDEX;
+import static org.apache.kafka.server.log.remote.storage.LocalTieredStorageEvent.EventType.FETCH_TRANSACTION_INDEX;
+import static org.apache.kafka.server.log.remote.storage.LocalTieredStorageEvent.EventType.OFFLOAD_SEGMENT;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.LEADER_EPOCH_CHECKPOINT;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.OFFSET_INDEX;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.PRODUCER_SNAPSHOT;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.SEGMENT;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.TIME_INDEX;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.RemoteLogSegmentFileType.TRANSACTION_INDEX;
+import static org.apache.kafka.server.log.remote.storage.RemoteLogSegmentFileset.openFileset;
+import static org.apache.kafka.server.log.remote.storage.RemoteTopicPartitionDirectory.openExistingTopicPartitionDirectory;
+import static org.apache.kafka.server.log.remote.storage.RemoteTopicPartitionDirectory.openTopicPartitionDirectory;
+
+/**
+ * An implementation of {@link RemoteStorageManager} which relies on the local file system to store
+ * offloaded log segments and associated data.
+ * <p>
+ * Due to the consistency semantic of POSIX-compliant file systems, this remote storage provides strong
+ * read-after-write consistency and a segment's data can be accessed once the copy to the storage succeeded.
+ * </p>
+ * <p>
+ * In order to guarantee isolation, independence, reproducibility and consistency of unit and integration
+ * tests, the scope of a storage implemented by this class, and identified via the storage ID provided to
+ * the constructor, should be limited to a test or well-defined self-contained use-case.
+ * </p>
+ * <p>
+ * The local tiered storage keeps a simple structure of directories mimicking that of Apache Kafka.
+ * <p>
+ * The name of each of the files under the scope of a log segment (the log file, its indexes, etc.)
+ * follows the structure UuidBase64-FileType.
+ * <p>
+ * Given the root directory of the storage, segments and associated files are organized as represented below.
+ * </p>
+ * <code>
+ * / storage-directory  / LWgrMmVrT0a__7a4SasuPA-0-topic / bCqX9U--S-6U8XUM9II25Q.log
```

Comment:
Is there any reason to do not prefix by topic-partition, and then by offset to make the resulting structure easier to navigate?

Reply:
Topics with the same name can be recreated with different Uuid. Currently, the RemoteStorageManger exposes API only to delete the segments, not the partition. When a topic gets deleted, there will be empty partition directories. To differentiate the stale topic directory from the active one, we need Uuid in the directory name.

Renaming the directory to `topic-partition-uuid`, requires a good amount of change. We can do it in the follow-up PR if required.

Reply:
> To differentiate the stale topic directory from the active one, we need Uuid in the directory name.

Sure, I wasn't advocating to have a partition directory, but something like this: 
```suggestion
 * / storage-directory  / topic-0-LWgrMmVrT0a__7a4SasuPA / 0000000000-bCqX9U--S-6U8XUM9II25Q.log
```

I agree changing the file structure may take a bit more, but the directory should be doable. I just added some suggestions on the 3 places that need changes if we want to update the layout with:

> In case we want to adopt the new structure:

on the comment. Let me know if this may be enough to improve the structure.

Reply:
Updated the directory structure to `topic-partition-uuid`.